ublox/src/ubx_packets/packets.rs

```diff
@@ -402,6 +402,137 @@ enum NavStatusFlags2 {
     Inactive = 3,
 }
 
+/// Information message conifg
+#[ubx_packet_recv_send]
+#[ubx(
+    class = 0x06,
+    id = 0x2,
+    fixed_payload_len = 10,
+    flags = "default_for_builder"
+)]
+struct CfgInf{
+    protocol_id: u8,
+    reserved: [u8; 3],
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_0: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_1: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_2: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_3: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_4: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_5: u8,
+}
+
+#[ubx_extend_bitflags]
+#[ubx(from, into_raw, rest_reserved)]
+bitflags! {
+    /// `CfgInfMask` parameters bitmask
+    #[derive(Default)]
+    pub struct CfgInfMask: u8 {
+        const ERROR= 0x1;
+        const WARNING = 0x2;
+        const NOTICE  = 0x4;
+        const TEST  = 0x8;
+        const DEBUG  = 0x10;
+    }
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x0,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfError{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x2,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfNotice{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x3,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfTest{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x1,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfWarning{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x4,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfDebug{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+
+mod inf {
+    pub(crate) fn convert_to_str(bytes: &[u8]) -> &str {
+        core::str::from_utf8(bytes).unwrap()
+    }
+
+    pub(crate) fn is_valid(bytes: &[u8]) -> bool {
+        true
+    }
```

Comment:
Panicking isn't an ideal approach (although I can't claim that this crate is currently panic-free), since it limits the user's options in the case of corrupt data. Would it be possible to fill in `is_valid` like so:
```
core::str::from_utf8(bytes).is_ok()
```
?

If performance is an issue, we could make `convert_to_str` return an option:
```
core::str::from_utf8(bytes).ok()
```
and set `map_type` to `Option<&str>`. (and also remove the `may_fail`, I think)

Reply:
I've fixed this by substituting an error string if UTF8 conversion fails.

Reply:
Playing around with it a bit locally, I'm a little worried about the fact that the user has no type information to indicate whether decoded succeeded (after all, in principle, the device could send a log message "** Invalid UTF8 string in UBX-INF packet **")

So, I'm thinking we can do this:
```
#[ubx_packet_recv]
#[ubx(
    class = 0x4,
    id = 0x4,
    max_payload_len = 1240,
    flags = "default_for_builder"
)]
struct InfDebug{
    #[ubx(map_type = Option<&str>,
        from = inf::convert_to_str,
        get_as_ref)]
    message: [u8; 0]
}
```
and:
```
mod inf {
    pub(crate) fn convert_to_str(bytes: &[u8]) -> Option<&str> {
        match core::str::from_utf8(bytes) {
            Ok(msg) => Some(msg),
            Err(_) => None,
        }
    }
}
```
and then, for the user, we get to keep the same API (they can get their message with `packet.message().unwrap()`)